### PR TITLE
446 add vde mode support

### PR DESCRIPTION
**Pull request by rpoletaev**

### CODE REVIEW

File: packages/api/route.go [hunk 1]
```diff
@@ -81,6 +78,14 @@ func Route(route *hr.Router) {
 	post(`test/:name`, ``, getTest)
 	post(`content`, `template:string`, jsonContent)
 
+	if !*conf.IsVDEMode {
```
Comment (ghost) [inline packages/api/route.go:76]: why ecosystemparam route is allowed in VDE mode? The same thing about systemparams route

File: packages/smart/funcs.go [hunk 2]
```diff
@@ -512,7 +512,7 @@ func DBSelect(sc *SmartContract, tblname string, columns string, id int64, order
 		ecosystem = sc.TxSmart.EcosystemID
 	}
 	tblname = GetTableName(sc, tblname, ecosystem)
-	if sc.VDE && *conf.CheckReadAccess {
+	if sc.VDE && *conf.IsVDEMode {
```
Comment (ghost) [inline packages/smart/funcs.go]: what does it mean "if VDE && IsVDE"?

File: packages/template/funcs.go [hunk 3]
```diff
@@ -461,7 +461,7 @@ func dbfindTag(par parFunc) string {
 	}
 	sc := par.Workspace.SmartContract
 	tblname := smart.GetTableName(sc, strings.Trim(converter.EscapeName((*par.Pars)[`Name`]), `"`), state)
-	if sc.VDE && *conf.CheckReadAccess {
+	if sc.VDE && *conf.IsVDEMode {
```
Comment (ghost) [inline packages/template/funcs.go]: the same as above

**Review by potehinre — APPROVED** [review]

File: packages/api/route.go (hunk 1 above)
Comment (ghost) [reply packages/api/route.go:76]: be more attentive please. Should it work in vde mode?
get(`ecosystems`, ``, authWallet, ecosystems)

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by ghost — APPROVED** [review]

**Pull request closed without merging**